extensions/amp-story-auto-ads/0.1/amp-story-auto-ads.js

```diff
@@ -22,7 +22,10 @@ import {
   StateChangeType,
 } from '../../amp-story/1.0/navigation-state';
 import {StateProperty} from '../../amp-story/1.0/amp-story-store-service';
+import {CSS as attributionCSS}
+  from '../../../build/amp-story-auto-ads-attribution-0.1.css';
 import {createElementWithAttributes, isJsonScriptTag} from '../../../src/dom';
+import {createShadowRootWithStyle} from '../../amp-story/1.0/utils';
```

Comment:
Maybe we should move this to `src/`?  Maybe `src/dom.js`?  @jridgewell would that make sense?

Reply:
I don't know of any other component that uses shadow roots.

Reply:
Well, this is now two: `amp-story` and `amp-story-auto-ads`.

We would also ideally refactor `amp-story-consent`, `amp-story-access`, and `amp-story-subscriptions` to be separate extensions.

Reply:
Sure, that's fine then.

Reply:
Sounds good. I will do this in a follow up PR, since I would like to get this into canary. Added a TODO